core/src/main/scala/akka/persistence/r2dbc/internal/BySliceQuery.scala

```diff
@@ -303,11 +305,30 @@ import org.slf4j.Logger
           throw new IllegalArgumentException(
             s"Unexpected offset [$offset] before latestBacktracking [${state.latestBacktracking}].")
 
-        state.copy(latestBacktracking = offset, rowCount = state.rowCount + 1)
+        // TODO: Could be nicer to track the seen count in TimestampOffset itself, along with the latest seen map.
+        //       Either as a seen count directly, or as a range of seen sequence numbers for each persistence id.
+        val newSeenCount =
+          if (offset.timestamp == state.latestBacktracking.timestamp) state.latestBacktrackingSeenCount + 1 else 1
+
+        state.copy(
+          latestBacktracking = offset,
+          latestBacktrackingSeenCount = newSeenCount,
+          rowCount = state.rowCount + 1)
+
       } else {
         if (offset.timestamp.isBefore(state.latest.timestamp))
           throw new IllegalArgumentException(s"Unexpected offset [$offset] before latest [${state.latest}].")
 
+        if (log.isDebugEnabled()) {
+          if (state.latestBacktracking.seen.nonEmpty &&
+            offset.timestamp.isAfter(state.latestBacktracking.timestamp.plus(firstBacktrackingQueryWindow)))
+            log.debugN(
+              "{} next offset is outside the backtracking window, latestBacktracking: [{}], offset: [{}]",
+              logPrefix,
+              state.latestBacktracking,
+              offset)
+        }
```

Comment:
Debugging for when the normal query is getting too far ahead. Left it in as could be useful.

Only debugging, as this can still happen if there are no events for over the backtracking window (2 minutes), then when new events arrive they'll be processed by the normal query but won't be seen by backtracking until after the longer backtracking behind-current-time has passed. It's expected behaviour and not a problem in this case — the only unprocessed events are within behind-current-time (10 seconds) and there won't be missed events on restart.